madanalysis/IOinterface/job_reader.py

```diff
@@ -1,406 +1,484 @@
-                line=line[:index]
+            index = line.find("#")
+            if index != -1:
+                line = line[:index]
 
             # Treating line
-            line=line.lstrip()
-            line=line.rstrip()
-            words=line.split()
-            if len(words)==0:
+            line = line.lstrip()
+            line = line.rstrip()
+            words = line.split()
+            if len(words) == 0:
                 continue
-
             # decoding the file
-            if len(words)==1 and words[0][0]=='<' and words[0][-1]=='>':
-                if words[0].lower()=='<safheader>':
+            if len(words) == 1 and words[0][0] == "<" and words[0][-1] == ">":
+                if words[0].lower() == "<safheader>":
                     beginTag.activate()
-                elif words[0].lower()=='</safheader>':
+                elif words[0].lower() == "</safheader>":
                     beginTag.desactivate()
-                elif words[0].lower()=='<saffooter>':
+                elif words[0].lower() == "<saffooter>":
                     endTag.activate()
-                elif words[0].lower()=='</saffooter>':
+                elif words[0].lower() == "</saffooter>":
                     endTag.desactivate()
-                elif words[0].lower()=='<description>':
+                elif words[0].lower() == "<description>":
                     descriptionTag.activate()
-                elif words[0].lower()=='</description>':
+                elif words[0].lower() == "</description>":
                     descriptionTag.desactivate()
-                elif words[0].lower()=='<statistics>':
+                elif words[0].lower() == "<statistics>":
                     statisticsTag.activate()
-                elif words[0].lower()=='</statistics>':
+                elif words[0].lower() == "</statistics>":
                     statisticsTag.desactivate()
-                elif words[0].lower()=='<data>':
+                elif words[0].lower() == "<data>":
                     dataTag.activate()
-                elif words[0].lower()=='</data>':
+                elif words[0].lower() == "</data>":
                     dataTag.desactivate()
-                elif words[0].lower()=='<histo>':
+                elif words[0].lower() == "<histo>":
                     histoTag.activate()
-                elif words[0].lower()=='</histo>':
+                elif words[0].lower() == "</histo>":
                     histoTag.desactivate()
                     plot.histos.append(copy.copy(histoinfo))
                     plot.histos[-1].positive.array = data_positive[:]
                     plot.histos[-1].negative.array = data_negative[:]
                     histoinfo.Reset()
+                    if multiweight_histo.is_consistent:
+                        plot.multiweight_histos.append(copy.deepcopy(multiweight_histo))
+                        print(multiweight_histo)
+                        print(multiweight_histo.shape)
```

Comment:
As a reminder, there are some print statements to be removed at the end.

Reply:
Thanks I'll definitely forget them there :D